Shopify/ruby-lsp
# Add workspace symbol

### Motivation

Closes #204 

Add support for the workspace symbol feature. This allows the index to be fuzzy searched by users, allowing them to search for classes, modules or constants defined anywhere including gems.

### Implementation

The implementation basically invokes `Index#fuzzy_search` and then transforms the result into the expected format for the LSP. We use the same trick as we do in definition here to avoid duplicating results for Sorbet users.

### Automated Tests

Since this feature is workspace wide and not related to a single file, I don't think it makes sense to add an expectations style test. I added a regular unit test.

### Manual Tests

1. Enable `rubyLsp.enabledFeatures.workspaceSymbol` in your VS Code settings
2. Launch the LSP on this branch
3. Click CMD + T to start workspace symbol
4. Type some entity name (e.g.: `YARP::CallNode`)
5. Verify you get results from Sorbet and the Ruby LSP (RBI and gem source)
